sections/06_discussion.md

```diff
@@ -183,32 +187,83 @@ multitudinous layers and architectures available
+[@doi:10.1038/ng.3830].
+
+The sharing of high-quality, labeled datasets will be especially valuable.  In
+addition, researchers who invest time to preprocess datasets to be suitable for
+deep learning can make the preprocessing code (e.g., Basset
+[@tag:Kelley2016_basset]) and cleaned data (e.g., MoleculeNet
+[@tag:Wu2017_molecule_net]) publicly available to catalyze further research. As
+discussed in the Categorize section, there are complex privacy and legal issues
+involved in sharing patient data and deep learning models trained with it but
+also recent techniques to mitigate these concerns.  Furthermore, in some
+domains, some of the best training data has been generated privately, for
+example, high-throughput chemical screening data at pharmaceutical companies.
+There is little expectation or incentive for this private data to be shared.
+`TODO: this last line is fairly pessimistic, is there something better to say?
+Is it possible to create the right incentives and/or protect proprietary data?`
+
+Code sharing and open source licensing is essential for continued progress in
+this domain.  Even though some journals have improved requirements about code
+availability at the time of publication, there remain many opportunities to do
+better.  Preprints are firmly established in many areas of machine learning and
+are becoming increasingly popular in the biomedical domain.  We advocate that
+code repositories are made available alongside preprints instead of waiting
+until publication, especially because some important work in this field is never
+published in a conference or journal.  In addition, it is important for authors
+to share not only code for their core models but also scripts and code used for
+data cleaning (see above) and hyperparameter optimization.  These improve
+reproducibility and serve as documentation of the detailed decisions that impact
+model performance but may not be exhaustively captured in a manuscript's methods
+text. `TODO: there are many other things that could be added here (e.g.
+licensing), but I'm not sure much time to spend on issues that are not deep
+learning-specific; maybe cite some good existing work?)`
+
+Because many deep learning models are often built using one of several popular
+deep learning frameworks, it is also possible to directly share trained
+predictive models.  The availability of pre-trained models can accelerate
+research, with image classifiers as an apt example.  A pre-trained neural
+network can be quickly fine-tuned on new data and used in transfer learning,
+both discussed below.  `TODO: bring up image-encoding for variant classification
+that used pre-trained image classifier?` "Model zoos", collections of
+pre-trained models, are not yet common in biomedical domains but have started to
+appear in genomics applications [@tag:Angermueller2016_single_methyl
+@tag:Dragonn]. `TODO: other examples, possibly from Categorize or Treat?`
```

Comment:
@c two paragraphs above I have "As discussed in the Categorize section, there are complex privacy and legal issues involved in sharing patient data and deep learning models trained with it but also recent techniques to mitigate these concerns."  Do you suggest I move that to this paragraph?